torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -620,174 +750,128 @@ class IrParser {
             TensorView* running_var = nullptr;
             if (!node->input(4)->type()->isSubtypeOf(
                     static_cast<c10::TypePtr>(NoneType::get()))) {
-              running_var =
+              running_mean =
                   value_map[node->input(4)->unique()]->as<TensorView>();
             }
 
+            TensorView* save_mean = nullptr;
+            // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+            if (!node->input(5)->type()->isSubtypeOf(
+                    static_cast<c10::TypePtr>(NoneType::get()))) {
+              // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+              save_mean = value_map[node->input(5)->unique()]->as<TensorView>();
+            }
+
+            TensorView* save_invstd = nullptr;
             // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
-            // auto training = constant_as<bool>(node->input(5));
-            // TORCH_INTERNAL_ASSERT(
-            //     training.has_value(),
-            //     "The training (bool) parameter is required.");
-            // const bool kTraining = training.value();
+            if (!node->input(6)->type()->isSubtypeOf(
+                    static_cast<c10::TypePtr>(NoneType::get()))) {
+              save_invstd =
+                  // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+                  value_map[node->input(6)->unique()]->as<TensorView>();
+            }
 
             // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
-            // auto momentum = constant_as<float>(node->input(6));
-            // TORCH_INTERNAL_ASSERT(
-            //     momentum.has_value(),
-            //     "The momentum (float) parameter is required.");
-            // const float kMomentum = momentum.value();
+            auto training = constant_as<bool>(node->input(7));
+            TORCH_INTERNAL_ASSERT(
+                training.has_value(),
+                "The training (bool) parameter is required.");
+            const bool kTraining = training.value();
 
             // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
-            auto eps = constant_as<float>(node->input(7));
+            auto eps = constant_as<float>(node->input(8));
             TORCH_INTERNAL_ASSERT(
                 eps.has_value(), "The EPS parameter is required.");
             const float kEps = eps.value();
 
-            // TODO: NAN when mean and variance are zero
-            // --ftz=true -- flush-to-zero
+            // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+            auto out_mask_list = constant_as<c10::List<bool>>(node->input(9));
```

Comment:
ditto.
This is actually a nitpick here, I think the mask would somehow always resolve as constants.